### ISSUE TITLE
macOS homebrew nightly build failed because of treesitter.c:134:3: error: implicit declaration of function 'ts_set_allocator' is invalid in C99 [-Werror,-Wimplicit-function-declaration] error

### ISSUE THREAD

**Issue by number5** [neovim/neovim#17658]
### Neovim version (nvim -v)

HEAD: 205b3765f

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

macOS 12.2.1 (ARM64)

### Terminal name/version

iTerm2 3.5.0beta4

### $TERM environment variable

xterm-256color

### Installation

homebrew build from source

### How to reproduce the issue

`brew unlink neovim && brew install --build-from-source --HEAD --force --fetch-HEAD neovim`

### Expected behavior

build should have been success and the new version of neovim installed.

### Actual behavior

```
==> Downloading http://www.leonerd.org.uk/code/libvterm/libvterm-0.1.4.tar.gz
######################################################################## 100.0%
==> Downloading https://github.com/libmpack/libmpack-lua/releases/download/1.0.8/libmpack-lua-1.0.8.t
==> Downloading from https://objects.githubusercontent.com/github-production-release-asset-2e65be/854
######################################################################## 100.0%
==> Downloading https://luarocks.org/manifests/gvvaughan/lpeg-1.0.2-1.src.rock
######################################################################## 100.0%
==> Cloning https://github.com/neovim/neovim.git
Updating /Users/bruce.wang/Library/Caches/Homebrew/neovim--git
==> Checking out branch master
Already on 'master'
Your branch is up to date with 'origin/master'.
HEAD is now at 205b3765f Merge pull request #17622 from dundargoc/refactor/clang-tidy/remove-redundant-casts
==> luarocks make --lua-dir=/opt/homebrew/opt/luajit-openresty --tree=/private/tmp/neovim-20220309-45
==> luarocks make --lua-dir=/opt/homebrew/opt/luajit-openresty --tree=/private/tmp/neovim-20220309-45
==> make install PREFIX=/private/tmp/neovim-20220309-4595-19ka89y/deps-build LDFLAGS=-static
==> cmake -S . -B build -DLIBLUV_LIBRARY=/opt/homebrew/opt/luv/lib/libluv.dylib
==> cmake --build build
Last 15 lines from /Users/bruce.wang/Library/Logs/Homebrew/neovim/05.cmake:
cd /tmp/neovim-20220309-4595-19ka89y/build/src/nvim && /opt/homebrew/Library/Homebrew/shims/mac/super/clang -DINCLUDE_GENERATED_DECLARATIONS -DMIN_LOG_LEVEL=3 -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -Dnvim_EXPORTS -I/tmp/neovim-20220309-4595-19ka89y/build/config -I/tmp/neovim-20220309-4595-19ka89y/src -I/tmp/neovim-20220309-4595-19ka89y/build/src/nvim/auto -I/tmp/neovim-20220309-4595-19ka89y/build/include -I/opt/homebrew/opt/luajit-openresty/include/luajit-2.1 -isystem /usr/local/include -isystem /tmp/neovim-20220309-4595-19ka89y/deps-build/include -isystem /opt/homebrew/opt/gettext/include -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_TS_HAS_SET_MATCH_LIMIT -DNVIM_TS_HAS_SET_ALLOCATOR -O2 -DNDEBUG -flto=thin -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=auto -MD -MT src/nvim/CMakeFiles/nvim.dir/mark.c.o -MF CMakeFiles/nvim.dir/mark.c.o.d -o CMakeFiles/nvim.dir/mark.c.o -c /tmp/neovim-20220309-4595-19ka89y/src/nvim/mark.c
/tmp/neovim-20220309-4595-19ka89y/src/nvim/lua/treesitter.c:134:3: error: implicit declaration of function 'ts_set_allocator' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  ts_set_allocator(xmalloc, xcalloc, xrealloc, xfree);
  ^
1 error generated.
gmake[2]: *** [src/nvim/CMakeFiles/nvim.dir/build.make:2533: src/nvim/CMakeFiles/nvim.dir/lua/treesitter.c.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
[100%] Building C object src/nvim/CMakeFiles/nvim.dir/marktree.c.o
cd /tmp/neovim-20220309-4595-19ka89y/build/src/nvim && /opt/homebrew/Library/Homebrew/shims/mac/super/clang -DINCLUDE_GENERATED_DECLARATIONS -DMIN_LOG_LEVEL=3 -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -Dnvim_EXPORTS -I/tmp/neovim-20220309-4595-19ka89y/build/config -I/tmp/neovim-20220309-4595-19ka89y/src -I/tmp/neovim-20220309-4595-19ka89y/build/src/nvim/auto -I/tmp/neovim-20220309-4595-19ka89y/build/include -I/opt/homebrew/opt/luajit-openresty/include/luajit-2.1 -isystem /usr/local/include -isystem /tmp/neovim-20220309-4595-19ka89y/deps-build/include -isystem /opt/homebrew/opt/gettext/include -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_TS_HAS_SET_MATCH_LIMIT -DNVIM_TS_HAS_SET_ALLOCATOR -O2 -DNDEBUG -flto=thin -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=auto -MD -MT src/nvim/CMakeFiles/nvim.dir/marktree.c.o -MF CMakeFiles/nvim.dir/marktree.c.o.d -o CMakeFiles/nvim.dir/marktree.c.o -c /tmp/neovim-20220309-4595-19ka89y/src/nvim/marktree.c
[100%] Building C object src/nvim/CMakeFiles/nvim.dir/math.c.o
cd /tmp/neovim-20220309-4595-19ka89y/build/src/nvim && /opt/homebrew/Library/Homebrew/shims/mac/super/clang -DINCLUDE_GENERATED_DECLARATIONS -DMIN_LOG_LEVEL=3 -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -Dnvim_EXPORTS -I/tmp/neovim-20220309-4595-19ka89y/build/config -I/tmp/neovim-20220309-4595-19ka89y/src -I/tmp/neovim-20220309-4595-19ka89y/build/src/nvim/auto -I/tmp/neovim-20220309-4595-19ka89y/build/include -I/opt/homebrew/opt/luajit-openresty/include/luajit-2.1 -isystem /usr/local/include -isystem /tmp/neovim-20220309-4595-19ka89y/deps-build/include -isystem /opt/homebrew/opt/gettext/include -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_TS_HAS_SET_MATCH_LIMIT -DNVIM_TS_HAS_SET_ALLOCATOR -O2 -DNDEBUG -flto=thin -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=auto -MD -MT src/nvim/CMakeFiles/nvim.dir/math.c.o -MF CMakeFiles/nvim.dir/math.c.o.d -o CMakeFiles/nvim.dir/math.c.o -c /tmp/neovim-20220309-4595-19ka89y/src/nvim/math.c
gmake[2]: Leaving directory '/private/tmp/neovim-20220309-4595-19ka89y/build'
gmake[1]: *** [CMakeFiles/Makefile2:540: src/nvim/CMakeFiles/nvim.dir/all] Error 2
gmake[1]: Leaving directory '/private/tmp/neovim-20220309-4595-19ka89y/build'
gmake: *** [Makefile:159: all] Error 2
```

`brew config` results:
```
HOMEBREW_VERSION: 3.4.1-22-g784be83
ORIGIN: https://github.com/Homebrew/brew
HEAD: 784be839b2c25c0d81c725a097dc83f2c37af418
Last commit: 4 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: a1d93f480d019a91137327d0eba470646d4fa8a5
Core tap last commit: 64 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_BROWSER: open
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.s8ble4kCzt/org.xquartz:0
HOMEBREW_EDITOR: nvim
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 13.0.0 build 1300
Git: 2.35.1 => /opt/homebrew/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.2.1-arm64
CLT: 13.2.0.0.1.1638488800
Xcode: N/A
Rosetta 2: false
```

`cmake --version`
```
cmake version 3.22.2
```
I have tried cmake 3.22.3. same errors.

My last successful build was with `HEAD-fdea157`

**Comment by clason** [comment]
Works fine here. This looks like a problem with your tree-sitter installation. Try reinstalling that.

**Comment by number5** [comment]
Tried removing and installing tree-sitter with HEAD and 0.20.6 few times, all got the same error.

Then I tried removing all the cache and the neovim check-outs and add option to keep the temp directory:
```
brew unlink neovim && brew install --build-from-source --HEAD --force --fetch-HEAD --keep-tmp neovim
```
Then it works. It coincided with a newer version of luajit (`luajit-openresty/2.1-20220309`) but I'm not sure that's related.

Anyway, I can't reproduce the error any more, will close this ticket
